src/JSONFormatter.cpp

```diff
@@ -0,0 +1,43 @@
+#include "JSONFormatter.hpp"
+#include "Log.hpp"
+
+namespace signal_estimator {
+
+void JSONFormatter::report_latency(double sw_hw, double hw, int sma_window, double hw_avg) {
+    if (first_output_) {
+        printf("[\n");
+        printf("  {\"sw+hw\": %7.3f, \"hw\": %7.3f, \"hw_avg%d\": %7.3f}", sw_hw,
```

Comment:
I suggest to remove limiting float precision from JSON formatter, since it's supposed to be used by programs, not humans, so shorter representation is non needed and even not desired.

Also let's replace `sw+hw` to something like `sw_hw` in JSON to make it a valid identifier, so that one could easily convert it to a class field in dynamic languages like python.